Realm/RLMApp.mm

```diff
@@ -128,25 +134,35 @@ - (instancetype)initWithBaseURL:(nullable NSString *)baseURL
         self.localAppName = localAppName;
         self.localAppVersion = localAppVersion;
         self.defaultRequestTimeoutMS = defaultRequestTimeoutMS;
-
-        _config.device_info.sdk = "Realm Swift";
-
-        // Platform info isn't available when running via `swift test`.
-        // Non-Xcode SPM builds can't build for anything but macOS, so this is
-        // probably unimportant for now and we can just report "unknown"
-        auto processInfo = [NSProcessInfo processInfo];
-        auto platform = [processInfo.environment[@"RUN_DESTINATION_DEVICE_PLATFORM_IDENTIFIER"]
-                         componentsSeparatedByString:@"."].lastObject;
-        RLMNSStringToStdString(_config.device_info.platform,
-                               platform ?: @"unknown");
-        RLMNSStringToStdString(_config.device_info.platform_version,
-                               [processInfo operatingSystemVersionString] ?: @"unknown");
-        RLMNSStringToStdString(_config.device_info.sdk_version, REALM_COCOA_VERSION);
+        RLMConfigureSyncConnectionParameters(_config);
         return self;
     }
     return nil;
 }
 
+static void RLMConfigureSyncConnectionParameters(realm::app::App::Config config) {
+    // Anonymized BundleId
+    NSString *bundleId = [[NSBundle mainBundle] bundleIdentifier];
+    NSData *bundleIdData = [bundleId dataUsingEncoding:NSUTF8StringEncoding];
+    RLMNSStringToStdString(config.device_info.bundle_id, RLMHashData(bundleIdData.bytes, bundleIdData.length));
```

Comment:
We are anonymizing the bundleId to match the reported anonymizedBundleId reported on the builder metrics.